pkg/compute/serve.go

```diff
@@ -0,0 +1,309 @@
+	progress.Step("Fetching latest Viceroy release...")
+
+	tmp, err := versioner.Download(context.Background(), latest)
+	if err != nil {
+		progress.Fail()
+		return fmt.Errorf("error downloading latest Viceroy release: %w", err)
+	}
+
+	if err := os.Rename(tmp, bin); err != nil {
+		if err := filesystem.CopyFile(tmp, bin); err != nil {
+			progress.Fail()
+			return fmt.Errorf("error moving latest Viceroy binary in place: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// updateViceroy checks if the currently installed version is out-of-date and
+// downloads the latest release from GitHub.
+func updateViceroy(verbose bool, progress text.Progress, version string, out io.Writer, versioner update.Versioner, latest semver.Version, bin string) error {
+	progress.Step("Checking installed Viceroy version...")
+
+	var installedViceroyVersion string
+
+	viceroyError := errors.RemediationError{
+		Inner:       fmt.Errorf("a Viceroy version was not found"),
+		Remediation: errors.BugRemediation,
+	}
+
+	// version output has the expected format: `viceroy 0.1.0`
+	segs := strings.Split(version, " ")
+
+	if len(segs) < 2 {
+		progress.Fail()
+		return viceroyError
+	}
+
+	installedViceroyVersion = segs[1]
+
+	if installedViceroyVersion == "" {
+		progress.Fail()
+		return viceroyError
+	}
+
+	current, err := semver.Parse(installedViceroyVersion)
+	if err != nil {
+		progress.Fail()
+
+		return errors.RemediationError{
+			Inner:       fmt.Errorf("error reading current version: %w", err),
+			Remediation: errors.BugRemediation,
+		}
+	}
+
+	if verbose {
+		text.Info(out, "Current Viceroy version: %s", current)
+	}
+
+	if latest.GT(current) {
+		// We already show the current Viceroy version when in Verbose mode, so if
+		// in non-verbose mode when dealing with an update (where we want to show
+		// both messages) we can omit the current version output.
+		if !verbose {
+			text.Info(out, "Current Viceroy version: %s", current)
+		}
+		text.Output(out, "Latest Viceroy version: %s", latest)
+
+		tmp, err := versioner.Download(context.Background(), latest)
+		progress.Step("Fetching latest Viceroy release...")
+		if err != nil {
+			progress.Fail()
+			return fmt.Errorf("error downloading latest Viceroy release: %w", err)
+		}
+		defer os.RemoveAll(tmp)
+
+		progress.Step("Replacing Viceroy binary...")
+
+		if err := os.Rename(tmp, bin); err != nil {
+			if err := filesystem.CopyFile(tmp, bin); err != nil {
+				progress.Fail()
+				return fmt.Errorf("error moving latest Viceroy binary in place: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// local spawns a subprocess that runs the compiled binary.
+func local(bin string, progress text.Progress, out io.Writer, env string, verbose bool) error {
+	if env != "" {
+		env = "." + env
+	}
+
+	wd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	manifest := filepath.Join(wd, fmt.Sprintf("fastly%s.toml", env))
```

Comment:
Am I interpreting this correctly that a `fastly compute serve --env test` will require a file named `fastlytest.toml` noting the lack of any separator between fastly and the stage? I think it would be a nicer ux if we add one:

```suggestion
	manifest := filepath.Join(wd, fmt.Sprintf("fastly-%s.toml", env))
```

Reply:
@phamann there's a `.` inbetween so `fastly.test.toml` look a couple lines up from that one I think.

Reply:
Lines 277-279

Reply:
Apologies, my fault, thats perfect 👌🏻